modules/core/src/nan_mask.simd.hpp

```diff
@@ -0,0 +1,435 @@
+
+        if (v_check_any(v_cmp_mask0) || v_check_any(v_cmp_mask1))
+        {
+            v_int32 v_dst0 = v_select(v_cmp_mask0, v_val, v_src0);
+            v_int32 v_dst1 = v_select(v_cmp_mask1, v_val, v_src1);
+
+            v_store(tptr + j, v_dst0);
+            v_store(tptr + j + cWidth, v_dst1);
+        }
+    }
+    vx_cleanup();
+#endif
+
+    for (; j < len; j++)
+    {
+        if ((tptr[j] & 0x7fffffff) > 0x7f800000)
+        {
+            tptr[j] = val.i;
+        }
+    }
+}
+
+
+static void patchNaNs_64f(uchar* ptr, size_t len, double newVal)
+{
+    int64_t* tptr = (int64_t*)ptr;
+    Cv64suf val;
+    val.f = newVal;
+
+    size_t j = 0;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+    v_int64 v_exp_mask = vx_setall_s64(0x7FF0000000000000);
+    v_int64 v_pos_mask = vx_setall_s64(0x7FFFFFFFFFFFFFFF);
+    v_int64 v_val = vx_setall_s64(val.i);
+
+    size_t cWidth = (size_t)VTraits<v_int64>::vlanes();
+    for (; j + cWidth*2 <= len; j += cWidth*2)
+    {
+        v_int64 v_src0 = vx_load(tptr + j);
+        v_int64 v_src1 = vx_load(tptr + j + cWidth);
+
+        v_int64 v_cmp_mask0 = v_lt(v_exp_mask, v_and(v_src0, v_pos_mask));
+        v_int64 v_cmp_mask1 = v_lt(v_exp_mask, v_and(v_src1, v_pos_mask));
+
+        if (v_check_any(v_cmp_mask0) || v_check_any(v_cmp_mask1))
+        {
+            // v_select is not available for v_int64, emulating it
+            v_int64 v_dst0 = v_or(v_and(v_cmp_mask0, v_val), v_and(v_not(v_cmp_mask0), v_src0));
+            v_int64 v_dst1 = v_or(v_and(v_cmp_mask1, v_val), v_and(v_not(v_cmp_mask1), v_src1));
+
+            v_store(tptr + j, v_dst0);
+            v_store(tptr + j + cWidth, v_dst1);
+        }
+    }
+    vx_cleanup();
+#endif
+
+    for (; j < len; j++)
+        if ((tptr[j] & 0x7FFFFFFFFFFFFFFF) > 0x7FF0000000000000)
+            tptr[j] = val.i;
+}
+
+PatchNanFunc getPatchNanFunc(bool isDouble)
+{
+    return isDouble ? (PatchNanFunc)GET_OPTIMIZED(patchNaNs_64f)
+                    : (PatchNanFunc)GET_OPTIMIZED(patchNaNs_32f);
+}
+
+////// finiteMask //////
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+
+template <typename _Tp, int cn>
+int finiteMaskSIMD_(const _Tp *src, uchar *dst, size_t total);
+
+template <>
+int finiteMaskSIMD_<float, 1>(const float *fsrc, uchar *dst, size_t total)
+{
+    const uint32_t* src = (const uint32_t*)fsrc;
+    const int osize = VTraits<v_uint8>::vlanes();
+    v_uint32 vmaskExp = vx_setall_u32(0x7f800000);
+
+    int i = 0;
+    for(; i <= (int)total - osize; i += osize )
+    {
+        v_uint32 vv0, vv1, vv2, vv3;
+        vv0 = v_ne(v_and(vx_load(src + i              ), vmaskExp), vmaskExp);
+        vv1 = v_ne(v_and(vx_load(src + i +   (osize/4)), vmaskExp), vmaskExp);
+        vv2 = v_ne(v_and(vx_load(src + i + 2*(osize/4)), vmaskExp), vmaskExp);
+        vv3 = v_ne(v_and(vx_load(src + i + 3*(osize/4)), vmaskExp), vmaskExp);
+
+        v_store(dst + i, v_pack_b(vv0, vv1, vv2, vv3));
+    }
+    vx_cleanup();
```

Comment:
> `vx_cleanup();`

Should be used in the end of externaly exposed functions only.

We see here internal template implementation code.

---

@vpisarev Direct usage of `v*_cleanup()` is anti-pattern.

`CV_INSTRUMENT_REGION()` / `CV_AVX_GUARD` / `VZeroUpperGuard`  should be used instead in externally exposed functions.

Currently we see:
- useless use: no reason to put that in the random middle of the code
- call is missing where it is really required.

Reply:
Removed `vx_cleanup()`